1_wqp_pull/src/wqp_inventory.R

```diff
@@ -71,86 +71,111 @@ inventory_wqp <- function(state_id, wqp_pull_params) {
-    state_counties <- filter(dataRetrieval::countyCd, STATE %in% gsub('US:', '', state_id))
-    
-    # split counties in half and make a 2-part call
-    split <- ceiling(nrow(state_counties)/2)
-    
-    # first half of counties
-    wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[1:split]
-    temp1 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
-    
-    # second half of counties
-    wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[(split+1):nrow(state_counties)]
-    temp2 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
-    
-    # combine outputs
-    temp_dat <- list(time = temp1$time + temp2$time,
-                     nrow = nrow(temp1$out) + nrow(temp2$out),
-                     out = bind_rows(temp1$out, temp2$out))
-  }
-  
+  # if(!'list' %in% class(temp_dat)) {
+  #   message('State call failed, calling by 1/2 of counties at a time.')
+  #   state_counties <- filter(dataRetrieval::countyCd, STATE %in% gsub('US:', '', state_id))
+  #
+  #   # split counties in half and make a 2-part call
+  #   split <- ceiling(nrow(state_counties)/2)
+  #
+  #   # first half of counties
+  #   wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[1:split]
+  #   temp1 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
+  #
+  #   # second half of counties
+  #   wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[(split+1):nrow(state_counties)]
+  #   temp2 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
+  #
+  #   # combine outputs
+  #   temp_dat <- list(time = temp1$time + temp2$time,
+  #                    nrow = nrow(temp1$out) + nrow(temp2$out),
+  #                    out = bind_rows(temp1$out, temp2$out))
+  # }
+
   # summarize what was retrieved and how long it took
   message('Retrieved ', temp_dat$nrow, ' rows of data in ', ceiling(temp_dat$time), ' seconds.')
-  
+
   return(temp_dat$out)
 }
 combine_inventory <- function(ind_file, ...) {
   dat_out <- bind_rows(...)
-  
+
   # keep columns of interest, put back lat/long
-  # keeping lat/long here in case there was a situation where you'd want to 
+  # keeping lat/long here in case there was a situation where you'd want to
   # take stock/map sites before pulling data
   dat_out <- dat_out %>%
-    select(OrganizationIdentifier, MonitoringLocationIdentifier, ResolvedMonitoringLocationTypeName, 
-           StateName, CountyName, HUCEightDigitCode, latitude = lat, longitude = lon, resultCount)
+    select(OrganizationIdentifier, MonitoringLocationIdentifier, ResolvedMonitoringLocationTypeName,
+           StateName, CountyName, HUCEightDigitCode, latitude = lat,
+           longitude = lon, resultCount) %>%
+    group_by(MonitoringLocationIdentifier) %>%
+    mutate(n_resultCount = sum(resultCount))
```

Comment:
Changing this column name will require me to change the downstream code that relies on this value. Can we keep it the same? 

Also, I think the reason you mutated instead of summarized is so that you can keep the other columns. However, in this current iteration, the same site will still be represented by more than one column. There are a few options here, but I think the easiest might be to: 

- assuming all values of the other columns are the same, you can add a distinct() call at the end of this. To test if our assumption is true that all metadata for a site is the same in each row the site is represented by, make sure that `nrow(dat_out)` and `length(unique(dat_out$MonitoringLocationIdentifier` are the same length.

Reply:
And note, you can only use the `distinct()` if you overwrite the column name with `mutate(resultCount = sum(resultCount))`

Reply:
To test that the assumption is true, that all metadata for a site is the same in each row the site is represented. I place a `Browse()`  call within the `combine_inventory` function and I checked if the `nrow(dat_out)` equals the `length(unique(dat_out$MonitoringLocationIdentifier))`


**R Output:**
Browse[1]> nrow(dat_out) == length(unique(dat_out$MonitoringLocationIdentifier))
[1] TRUE